WasteWar/Assets/Scripts/Drawing/StructurePlacement.cs

```diff
@@ -0,0 +1,70 @@
+﻿using UnityEngine;
+using Constants;
+
+public class StructurePlacement : MonoBehaviour
+{
+    [SerializeField]
+    private Terrain terrain;
+    [SerializeField]
+    private GameObject drawingScripts;
+    
+    private RaycastHit hit;
+    private Ray ray;
+
+    private StructureGrid structureGrid;
+    private Vector3 terrainSize;
+    DrawOnTerrain terrainCanvas;
+
+    private bool isAStructureSelected;
+    bool isSpaceOccupied = false;
+
+    void Start()
+    {
+        terrainCanvas = drawingScripts.GetComponent<DrawOnTerrain>();
+        terrainSize = terrain.GetComponent<Terrain>().terrainData.size;
+        structureGrid = new StructureGrid(terrainSize.x,terrainSize.z, GridConstants.Instance.FloatCellSize());
+    }
+
+    void Update() {
+        ray = Camera.main.ScreenPointToRay(Input.mousePosition);
```

Comment:
it's considered best practice to reference the camera rather than using Camera.main